### PR TITLE
Add Support to ES 5 for Additional Geo Shapes

### PR DESCRIPTION
**Pull request by mwizeman**
Add support to the elasticsearch5 module to both store and search with more of the ElasticSearch geo shapes. This PR adds support for the following shapes:
- GeoRect
- GeoCircle
- GeoLine
- GeoPolygon
- GeoCollection

In addition, this PR adds DISJOINT, INTERSECT, and CONTAINS geo search types.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,8 @@
 * Changed: Removed support for NameSubstitutionStrategy from the Elasticsearch modules. NameSubstitutionStrategy is still available for Accumulo modules
 * Changed: Added property checking when using the `has` method to query the graph for a value. An exception will now be thrown if an attempt is made to either query a property that does not exist or to text query a property that isn't full text indexed
 * Note: Upgrading to this version will require re-indexing if you use NameSubstitutionStrategy or are switching to the ES 5 module. 
-
+* ADDED: Support for extended GeoShape storage and search. Currently the new shapes are only supported by the ElasticSearch 5 module.
```
Comment (joeferner) [inline CHANGELOG.md]: Can you fix the case "Added" not "ADDED"

File: core/src/main/java/org/vertexium/type/GeoCollection.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    public GeoCollection(List<GeoShape> geoShapes, String description) {
+        super(description);
+        this.geoShapes.addAll(geoShapes);
+    }
+
+    @Override
+    public boolean intersects(GeoShape geoShape) {
+        throw new VertexiumException("Not implemented for argument type " + geoShape.getClass().getName());
```
Comment (joeferner) [inline core/src/main/java/org/vertexium/type/GeoCollection.java]: shouldn't you just be able to call intersects on all `geoShapes` here?

File: core/src/main/java/org/vertexium/type/GeoCollection.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    @Override
+    public boolean within(GeoShape geoShape) {
+        throw new VertexiumException("Not implemented for argument type " + geoShape.getClass().getName());
```
Comment (joeferner) [inline core/src/main/java/org/vertexium/type/GeoCollection.java]: same here

File: elasticsearch5/src/main/java/org/vertexium/elasticsearch5/ElasticsearchSearchQueryBase.java [hunk 4]
```diff
@@ -602,76 +604,75 @@ protected QueryBuilder getFilterForGeoComparePredicate(GeoCompare compare, HasVa
...
+            }
+
+            ShapeBuilder shapeBuilder = getShapeBuilder(value);
+            ShapeRelation relation = ShapeRelation.getRelationByName(compare.getCompareName());
+            filters.add(new GeoShapeQueryBuilder(propertyName, shapeBuilder).relation(relation));
         }
         return getSingleFilterOrOrTheFilters(filters, has);
     }
 
+    private ShapeBuilder getShapeBuilder(GeoShape geoShape) {
```
Comment (joeferner) [inline elasticsearch5/src/main/java/org/vertexium/elasticsearch5/ElasticsearchSearchQueryBase.java]: can you split this method up, per type. It looks very dense.

**Comment by mwizeman** [comment]
@joeferner I addressed all of your comments with 46e9ff8de96cb4be21dc06bad995c444e25d12d0. Thanks for taking a look at it.

**Pull request merged**